Upload exercise 4  of class 5 homework. Dynamic version.

lucilakim opened the pull request:
These files calculate values from a list of numbers

lucilakim committed:
Upload exercise 4  of class 5 homework. Dynamic version.

These files calculate values from a list of numbers

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_index.html:
```diff
@@ -0,0 +1,35 @@
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <link rel="stylesheet" href="task-4_styles.css">
```
Por lo general en javascript los archivos se escriben en ``kebab-case``.

lucilakim replied:
Serían todos los nombres de los archivos? Me entró la duda porque es un css, dentro de un html. 
Los cambio a todos por las dudas

Genebson replied:
Si, todos los archivos

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+const $bodyFromHtml = document.querySelector('body');
```
Acá podes usar ``$body`` sin aclarar que viene del HTML porque no es el mismo caso que el anterior que tenías 2 o 3 variables que eran numbers/numbersFromHTML/etc

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+
+calculateValuesButton.onclick = () => {
+    const $resultsOfList = document.querySelector("#results-of-the-calculations-of-the-list");
+    const valuesQuantity = document.querySelectorAll('.input-values');
```
Estos serían los numeros no? Acá podrías copiar los nombres de las variables del ejercicio anterior numbersFromHTML

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+calculateValuesButton.onclick = () => {
+    const $resultsOfList = document.querySelector("#results-of-the-calculations-of-the-list");
+    const valuesQuantity = document.querySelectorAll('.input-values');
+    const values = getNumbers(valuesQuantity);
```
Y acá numbers, cómo [acá](https://github.com/lucilakim/r-argentina-programa/compare/main...task-5_4_getLiValues#diff-6d472a3b83a5e4a0af3052f4e688ba05678f8a0fe30f3ecced652ee4e00fe891R2)

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+}
+
+function createInputItems(quantity) {
+    const formTitle = document.createElement('h3');
```
Acordate de no aclarar el elemento HTML en las variables, ``title`` queda mejor

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+    const resultCalculation = document.querySelector("#results-of-the-calculations-of-the-list")
+    const list = document.createElement('ol');
+    const title = document.createElement('h4');
+    title.innerText = `List`;
```
Acá no es necesario usar backticks no?

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+    const $resultsOfList = document.querySelector("#results-of-the-calculations-of-the-list");
+    const title = document.createElement('h4');
+    title.innerText = 'Results'
+    const firtsResult = document.querySelector("#first-result");
```
Acá hay un typo ``firts``

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+}
+
+
+function setResultOfList() {
```
``setResult`` está bien, cuanto más generico mejor, si el día de mañana deja de ser una lista y es un div o el elemento que sea, tenes que cambiar el nombre de la función, el elemento, etc

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_script.js:
```diff
@@ -0,0 +1,167 @@
+    $resultsOfList.insertBefore(title, firtsResult)
+}
+
+function getValuesFrom(inputList) {
```
getNumbers(numbers)?

lucilakim replied:
La tengo en la linea 104 así y habría dos funciones con la misma firma, no hay problema?

La de la 96 obtiene valores de los input que ingresa el usuario
(Cómo el ejercicio pedía que sean los valores del li, esos valores los input los obtengo, los pongo en los li y después en la función de la 104 rescato los valores de ahí.)

La 104 obtiene los valores de los li, era la que me habías pasado.

![image](https://user-images.githubusercontent.com/91380597/205126980-adc6e7a5-848b-4ad9-a561-1911d4553c76.png)

Genebson replied:
Tenes razón, no había visto esa función. Al parámetro si le podes dar una vuelta de tuerca para sacarle la palabra ``List``.

Genebson commented on introduccion-a-js-master/tareas/clase-5/task-4_styles.css:
```diff
@@ -0,0 +1,34 @@
+.list-value {
+    margin-left: 5px;
+}
+.calculate-list-values-button {
+    display: none;
+}
+
+.results-of-the-calculations-of-the-list {
+    display: none;
+}
```
Acá podes aglomerar los dos selectores así:
``.calculate-list-values-button,
.results-of-the-calculations-of-the-list  {
    display: none;
}
``

lucilakim committed:
Correct names, message and board selectors css.

For better reading and code optimization.

Correct file, variables, functions, and parameters names.

Optimize message and match css selectors with repeating properties.
